django/contrib/contenttypes/management.py

```diff
@@ -1,9 +1,65 @@
 from django.apps import apps
-from django.db import DEFAULT_DB_ALIAS, router
+from django.core.management.color import color_style
+from django.db import DEFAULT_DB_ALIAS, migrations, router, transaction
+from django.db.utils import IntegrityError
 from django.utils import six
 from django.utils.six.moves import input
 
 
+class RenameContentType(migrations.RunPython):
+    def __init__(self, app_label, old_name, new_name, verbosity):
+        self.app_label = app_label
+        self.old_name = old_name
+        self.new_name = new_name
+        self.verbosity = verbosity
+        super(RenameContentType, self).__init__(self.rename_forward, self.rename_backward)
+
+    def _rename(self, apps, old_name, new_name):
+        ContentType = apps.get_model('contenttypes', 'ContentType')
+        try:
+            content_type = ContentType.objects.get_by_natural_key(self.app_label, old_name.lower())
+        except ContentType.DoesNotExist:
+            pass
+        else:
+            content_type.model = new_name.lower()
+            using = router.db_for_write(ContentType, instance=content_type)
+            try:
+                with transaction.atomic(using=using):
+                    content_type.save(update_fields={'model'})
+            except IntegrityError:
+                # Gracefully fallback if a stale content type causes a conflict.
+                if self.verbosity:
+                    print(color_style().WARNING(
+                        "Failed to rename content type '%(app_label)s.%(old_name)s' "
+                        "to '%(app_label)s.%(new_name)s'.\nYou might have stale content types in conflict." % {
+                            'app_label': self.app_label,
+                            'old_name': old_name,
+                            'new_name': new_name,
+                        }
+                    ))
```

Comment:
I'm just thinking if we can reduce the chance of Django leaving behind or creating stale content types, then this should be less of an issue. I think option 2/3 of prompting the user or baling out when we're not sure would be okay. I expect/hope this would mostly be encountered in local development scenarios where interactive is applicable.